### PR TITLE
fix: Date picker displays the previous date as selected by the user

### LINKED ISSUE #983: DatePicker's highlighted date is set to current date even after user selects a different date.
**Summary:**

DatePicker's highlighted date is set to current date even after user selects a different date.

**Steps to reproduce:**
Goto->Accounts->Loan fragment
Choose floating action button to apply for loan
Set expected disbursement date to future value and open datepicker dialog again.

**Expected behaviour:**

DatePickers highlighted date should be the date selected by user previously

**Observed behaviour:**

DatePickers highlighted date is current date always

**Device and Android version:**

Motorola Moto G4 (Android 7.0,API 24)
**Screenshots:**

ezgif com-video-to-gif

Can be created by pressing the Volume Down and Power Button at the same time on Android 4.0 and higher.

### PR DESCRIPTION
**Pull request by abhin2149**
issue:#983

Fixes #983

Date picker now displays the previous date as selected by the user

### CODE REVIEW

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 1]
```diff
@@ -14,10 +14,10 @@
 import android.widget.DatePicker;
 
 import org.mifos.mobilebanking.R;
-
 import java.util.Calendar;
 import java.util.Date;
 
+
```
Comment (RajaVamsi11) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Remove this empty line. I think this line is not necessary.

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 2]
```diff
@@ -67,6 +69,7 @@ public static MFDatePicker newInstance(Fragment fragment, int datePickerType) {
 
         mfDatePicker.setArguments(args);
 
+
```
Comment (RajaVamsi11) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Here also remove the empty line

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 3]
```diff
@@ -78,12 +81,17 @@ public static String getDatePickedAsString() {
...
         DatePickerDialog dialog = new DatePickerDialog(getActivity(),
                 R.style.MaterialDatePickerTheme,
                 this,
-                calendar.get(Calendar.YEAR),
-                calendar.get(Calendar.MONTH),
-                calendar.get(Calendar.DAY_OF_MONTH));
+                year,
+                month,
+                day);
+
```
Comment (RajaVamsi11) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Here too.

Comment (abhin2149) [reply app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: do i have to open a new pull request after changing or is there some way to do it in the same pull request only?

Comment (RajaVamsi11) [reply app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: There is a way to do in the same pull request.
After removing the extra lines (which I commented ) while committing just add `amend` method like this `git commit --amend` 
You can refer [this](https://es.atlassian.com/git/tutorials/rewriting-history#git-commit--amend) for more information

Comment (abhin2149) [reply app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: okay thanks

Comment (abhin2149) [reply app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: is the latest build fine?

Comment (RajaVamsi11) [reply app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Fine

**Comment by abhin2149** [comment]
@luckyman20 @miPlodder @therajanmaurya  Please review this PR and suggest changes if any.

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java [hunk 4]
```diff
@@ -97,6 +99,7 @@ public Dialog onCreateDialog(Bundle savedInstanceState) {
                 dialog.getDatePicker().setMaxDate(new Date().getTime());
                 break;
         }
+
```
Comment (shiv07tiwari) [inline app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: Irrelevant space . Kindly remove it @abhin2149

Comment (abhin2149) [reply app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: okay

**Review by RajaVamsi11 — COMMENTED** [review]
@abhin2149 Please remove merge conflicts also.

**Comment by abhin2149** [comment]
> 
> 
> @abhin2149 Please remove merge conflicts also.
how do i do it?

File: app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java (hunk 4 above)
Comment (RajaVamsi11) [reply app/src/main/java/org/mifos/mobilebanking/utils/MFDatePicker.java]: There will be an option to solve conflicts. Click it and you know where the files are conflicting.

**Comment by RajaVamsi11** [comment]
There will be an option to solve conflicts. Click it and you will know where the files are conflicting.

**Comment by abhin2149** [comment]
> 
> 
> There will be an option to solve conflicts. Click it and you will know where the files are conflicting.

I did but when i try to merge with development branch it says that the branch is up to date.

**Comment by abhin2149** [comment]
Also the Resolve conflicts button is greyed out

**Comment by prashantkh19** [comment]
@abhin2149 can you fix the merge conflicts? You can pull recent changes from the remote branch in android studio only and merge there.

**Comment by geekanamika** [comment]
As There's no updated after request from @prashantkh19 , Closing this PR & will keep this issue for GCI2019.

**Pull request closed without merging**